### PR TITLE
Add option on put to delete previous comments

### LINKED ISSUE #95: Option to delete previously made comments on a pull request
A use case we have is to make comments as a CI bot on a pull request, such as output of Terraform plan to know what will be applied if the pull request is merged.

Example pipeline snippet:
```
on_success:
      put: pull-request
      params
          path: pull-request
          status: success
          comment_file: github-comment/comment.txt
```

Comments can become outdated as changes are made to the pull request, since this verification runs on every commit. It would be nice to have a boolean option on `put` that would delete past comments made on the pull request by this resource. This would make sure only the latest information from CI is present on a pull request.

Example pipeline snippet:
```
on_success:
      put: pull-request
      params
          path: pull-request
          status: success
          comment_file: github-comment/comment.txt
          only_latest_comment: true
```

Wondering what thoughts are on a feature like this.

### PR DESCRIPTION
**Pull request by rcoy-v**
Resolves #95 

I would like to have an e2e test, but have not done that yet. Would like to get feedback first before that. Tested by building docker image and using in a pipeline against test pull request.

### CODE REVIEW

File: github.go [hunk 1]
```diff
@@ -278,6 +279,56 @@ func (m *GithubClient) UpdateCommitStatus(commitRef, baseContext, statusContext,
...
+		"commentsLast":    githubv4.Int(100),
+	}
+
+	if err := m.V4.Query(context.TODO(), &getComments, vars); err != nil {
+		return err
+	}
+
+	for _, e := range getComments.Repository.PullRequest.Comments.Edges {
+		if e.Node.Author.Login == getComments.Viewer.Login {
+			_, err := m.V3.Issues.DeleteComment(context.TODO(), m.Owner, m.Repository, e.Node.DatabaseId)
```
Comment (rcoy-v) [inline github.go]: I tried to use the graphql mutation `deleteIssueComment`, but was receiving an unhelpful error message. The v3 api worked immediately.

**Comment by itsdalmo** [comment]
Thanks for the PR @rcoy-v, I've been wanting this feature myself also 😅Will try to get a review ready tomorrow 🙌

**Comment by jeromelamang** [comment]
Hi

**Comment by rcoy-v** [comment]
Any thoughts on how you would like an e2e test for this? The current ones appear to be using an archived test repository. I don't think that will work here, as the test will need to add new comments on a pull request for setup then actually validate they were deleted.

**Comment by itsdalmo** [comment]
Hi, sorry for the late reply! 

> Any thoughts on how you would like an e2e test for this? The current ones appear to be using an archived test repository.

I see the problem with using an archived test repository, but I'm not entirely sure about how to do this. Perhaps we just need a new repository with a single PR that we can use to test side-effects (like adding/deleting comments) against?

**Comment by rcoy-v** [comment]
@itsdalmo I've added e2e tests. 

I created a new repo, https://github.com/rcoy-v/github-pr-resource-e2e. Each test creates a new pull request from a pre-configured branch of changes, and deletes afterwards.

I imagine you would want ownership of this. I think that I should be able to transfer ownership of the repo, and the e2e test `owner` would have to be updated to your username.

A test case not covered would be not deleting someone else's comments. This would require making a test GitHub user and adding comments from them using a second access token. This could be done, but I felt that would be up to you how to own a test user and more credentials. I've manually tested this works as expected.

**Comment by rcoy-v** [comment]
@itsdalmo anything I can do to help get this merged?

**Comment by itsdalmo** [comment]
Hi @rcoy-v - I'll give it a last review/merge this evening. Sorry I've been unresponsive (just tied up with unrelated stuff).

File: e2e/e2e_test.go [hunk 2]
```diff
@@ -3,15 +3,19 @@
 package e2e_test
 
 import (
+	"context"
+	"fmt"
+	github2 "github.com/google/go-github/github"
```
[the remark this answers is not in the document]
Comment (rcoy-v) [reply e2e/e2e_test.go]: It was colliding with a variable. Refactored.

**Pull request merged**